tests/functional/simple_snapshot/test_basic_snapshot.py

```diff
@@ -0,0 +1,323 @@
+
+    path = os.path.join(project.test_data_dir, "invalidate_postgres.sql")
+    project.run_sql_file(path)
+
+    path = os.path.join(project.test_data_dir, "update.sql")
+    project.run_sql_file(path)
+
+    results = run_dbt(["snapshot"])
+    assert len(results) == NUM_SNAPSHOT_MODELS
+
+    run_dbt(["test"])
+    table_comp.assert_tables_equal("snapshot_actual", "snapshot_expected")
+
+
+def ref_setup(project, NUM_SNAPSHOT_MODELS):
+
+    path = os.path.join(project.test_data_dir, "seed_pg.sql")
+    project.run_sql_file(path)
+    results = run_dbt(["snapshot"])
+    assert len(results) == NUM_SNAPSHOT_MODELS
+
+    results = run_dbt(["run"])
+    assert len(results) == 1
+
+
+# these fixtures are slight variations of each other for the basic snapshot tests run
+@pytest.fixture
+def basic_snapshot(project):
+    NUM_SNAPSHOT_MODELS = 1
+    table_comp = TableComparison(
+        adapter=project.adapter, unique_schema=project.test_schema, database=project.database
+    )
+
+    snapshot_setup(project, NUM_SNAPSHOT_MODELS, table_comp)
+
+
+@pytest.fixture
+def check_cols_snapshot(project):
+    NUM_SNAPSHOT_MODELS = 2
+    table_comp = OverrideTableComparison_dbt(
+        adapter=project.adapter, unique_schema=project.test_schema, database=project.database
+    )
+
+    snapshot_setup(project, NUM_SNAPSHOT_MODELS, table_comp)
+
+
+@pytest.fixture
+def revived_snapshot(project):
+    NUM_SNAPSHOT_MODELS = 2
+    table_comp = RevivedTableComparison(
+        adapter=project.adapter, unique_schema=project.test_schema, database=project.database
+    )
+
+    snapshot_setup(project, NUM_SNAPSHOT_MODELS, table_comp)
+
+
+@pytest.fixture
+def basic_ref(project):
+    NUM_SNAPSHOT_MODELS = 1
+    ref_setup(project, NUM_SNAPSHOT_MODELS)
+
+
+@pytest.fixture
+def basic_ref_two_snapshots(project):
+    NUM_SNAPSHOT_MODELS = 2
+    ref_setup(project, NUM_SNAPSHOT_MODELS)
+
+
+class Basic:
+    @pytest.fixture(scope="class")
+    def snapshots(self, snapshots_pg):  # noqa: F811
+        return snapshots_pg
+
+
+@pytest.mark.usefixtures("project")
+class TestBasicSnapshot(Basic):
+    def test_basic_snapshot(project, basic_snapshot):
+        basic_snapshot
+
+
+@pytest.mark.usefixtures("project")
+class TestBasicRef(Basic):
+    def test_basic_ref(project, basic_ref):
+        basic_ref
+
+
+class CustomNamespace:
+    @pytest.fixture(scope="class")
+    def snapshots(self, snapshots_pg_custom_namespaced):  # noqa: F811
+        return snapshots_pg_custom_namespaced
+
+    @pytest.fixture(scope="class")
+    def macros(self, macros_custom_snapshot):  # noqa: F811
+        return macros_custom_snapshot
+
+
+@pytest.mark.usefixtures("project")
+class TestBasicCustomNamespace(CustomNamespace):
+    def test_basic_snapshot(project, basic_snapshot):
+        basic_snapshot
+
+
+@pytest.mark.usefixtures("project")
+class TestRefCustomNamespace(CustomNamespace):
+    def test_basic_ref(project, basic_ref):
```

Comment:
If the mark.usefixtures("project") is stated for the whole calss, is the project fixture here still needed?